### PR TITLE
Add `awaitData` tests

### LINKED ISSUE #3818: Add tests for cursors
Add test for [this case](https://github.com/FerretDB/FerretDB/issues/3739#issuecomment-1833398536) [there](https://github.com/FerretDB/FerretDB/tree/main/integration/cursors).

Think of other cases we could add.

`maxTimeMS` / timeouts are in separate issues.

```[tasklist]
### Tasks
- [x] `AwaitData` tests
- [x] tailable=false, awaitData=true test
```

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #3818.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [ ] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by AlekSi** [comment]
> I set Reviewers ([@FerretDB/core](https://github.com/orgs/FerretDB/teams/core))

That wasn't done

File: integration/cursors/awaitdata_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		{"getMore", cursorID},
+		{"collection", collection.Name()},
+		{"batchSize", 1},
+		{"maxTimeMS", 1000 * 60 * 10},
+	}
+
+	go func() {
+		time.Sleep(2 * time.Second)
+		_, err = collection.InsertOne(ctx, bson.D{{"v", "bar"}})
+		require.NoError(t, err)
```
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:76]: `require` does not work correctly in a goroutine. It also may not work if the test ends before this line is reached.

We should use assert and wait for the goroutine to finish, or combine both and just send the error to the channel:

```go
ch := make(chan error)

go func() {
		ctxutil.Sleep(ctx, 2 * time.Second)
		_, err = collection.InsertOne(ctx, bson.D{{"v", "bar"}})
		ch <- err
}()

// ...

require.NoError(t, <-ch)
```

File: integration/cursors/awaitdata_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	var firstBatch *types.Array
+	firstBatch, cursorID := getFirstBatch(t, res)
+
+	require.Equal(t, 1, firstBatch.Len())
+
+	getMoreCmd := bson.D{
+		{"getMore", cursorID},
+		{"collection", collection.Name()},
+		{"batchSize", 1},
+		{"maxTimeMS", 1000 * 60 * 10},
```
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:70]: `(10 * time.Minute).Milliseconds()`

File: integration/cursors/awaitdata_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	tt.Parallel()
+
+	t := setup.FailsForFerretDB(tt, "https://github.com/FerretDB/FerretDB/issues/2283")
+
+	s := setup.SetupWithOpts(t, nil)
+
+	db, ctx := s.Collection.Database(), s.Ctx
+
+	opts := options.CreateCollection().SetCapped(true).SetSizeInBytes(10000000000)
+	err := db.CreateCollection(s.Ctx, t.Name(), opts)
```
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:41]: `testutil.CollectionName(t)`

**Review by henvic — APPROVED** [review]

**Review by henvic — DISMISSED** [review]

**Review by henvic — APPROVED** [review]

File: integration/cursors/awaitdata_test.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	}()
+
+	err = collection.Database().RunCommand(ctx, getMoreCmd).Decode(&res)
+	require.NoError(t, err)
+
+	require.NoError(t, <-insertChan)
+
+	nextBatch, nextID := getNextBatch(t, res)
+	require.Equal(t, cursorID, nextID)
+	require.Equal(t, 1, nextBatch.Len())
```
Comment (AlekSi) [inline integration/cursors/awaitdata_test.go:89]: Ok, now I don't understand why we use a goroutine at all. There is no blocking anywhere. `res` is not checked after `RunCommand` and before waiting for `insert` to complete. `getNextBatch` doesn't do any I/O. What are we testing there, exactly?

Comment (noisersup) [reply integration/cursors/awaitdata_test.go:89]: What's the difference between checking `res` before or after getting insertChan content? 

In this test we want to:
1) Run `getMore` command on tailable awaitData cursor, that has 10 minutes of maxTime.
2) After some time (1 second) insert data in parallel, so the previously called getMore command unblock and fetch the new document
3) validate that `getMore` batch has 1 document. If it had 0 it'd mean that cursor didn't block which is incorrect behavior

**Review by AlekSi — DISMISSED** [review]

**Review by henvic — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**